be/src/agent/task_worker_pool.cpp

```diff
@@ -250,13 +250,13 @@ void TaskWorkerPool::submit_tasks(std::vector<TAgentTaskRequest>* tasks) {
 }
 
 bool TaskWorkerPool::_register_task_info(const TTaskType::type task_type, int64_t signature) {
-    std::lock_guard task_signatures_lock(_s_task_signatures_lock);
+    std::lock_guard task_signatures_lock(_s_task_signatures_lock[task_type]);
```

Comment:
I don't quite understand how can this guarantee thread safety,  the '[]' operator of `std::map` is not a const method, which means it's not safe to call `_s_task_signatures[task_type]` from different threads, while each thread has a different value of `task_type`.

Reply:
Maybe you should declare `_s_task_signatures` as `std::set<int64_t> _s_task_signatures[TTaskType::type::TASK_TYPE_COUNT]`